src/utils/background_fetch.js

```diff
@@ -0,0 +1,126 @@
+//    parameters.
+//
+// **Returns:**
+//  - A string representing the complete URL with encoded search parameters.
+const searchParamsURL = ({ base, params }) => {
+  const url = new URL(base);
+  Object.keys(params).forEach((key) =>
+    url.searchParams.append(key, params[key])
+  );
+  return url.toString();
+};
+
+function triggerFetchRequest(url, options, sender, sendResponse) {
+  // Callback for Dispatching Background Fetch
+  // This internal function handles setting headers, logging information,
+  // dispatching the fetch request, and sending the response back
+  const dispatchCallback = (_url, _options) => {
+    console.debug(_url, _options);
+    _options['headers'] = { ...authHeader };
+    if (_options.method == 'GET') {
+      _options['headers'] = { ..._options['headers'], ...jsonHeader };
+    }
+    console.info(`RECAP: Dispatching ${_options.method} for ${_url}`);
+    // dispatch fetch and return the response
+    fetch(_url, _options)
+      .then((res) => res.json())
+      .then((json) => sendResponse(json))
+      .catch((err) => sendResponse({ error: err.message }));
+  };
+  // Check for existing body data and handle accordingly
+  if (!options.body) {
+    dispatchCallback(url, options);
+    return true;
+  }
+
+  // Handle body with HTML content
+  if (options.body.html) {
+    let blob = new Blob([options.body.html], { type: 'text/plain' });
+    options.body['filepath_local'] = blob;
+    delete options.body.html;
+  }
+
+  // Handle body without document
+  if (!options.body.document) {
+    const body = buildFormData({ ...options.body });
+    dispatchCallback(url, { ...options, body });
+    return true;
+  }
+
+  // This section handles scenarios where the request body contains a document
+  // that needs to be retrieved from the Chrome extension's local storage
+  // associated with the sender's tab ID.
+  chrome.storage.local.get(sender.tab.id.toString(), async (store) => {
+    // get the dataUrl for the file in storage
+    let storeKey = sender.tab.id.toString();
+    let file = store[storeKey]['pdf_blob'] || store[storeKey]['zip_blob'];
+    const blob = await fetch(file).then(
+      (res) => res.blob()
+    );
+    const body = buildFormData({ ...options.body, filepath_local: blob });
+    dispatchCallback(url, { ...options, body });
+  });
+  // return true to allow for the async function to complete
+  return true;
+}
+
+// This function dispatches fetch requests based on the received action and
+// data. It constructs the appropriate URL and options for each action, and
+// then triggers the fetch request using the `triggerFetchRequest` function.
+export const handleBackgroundFetch = (req, sender, sendResponse) => {
+  const { action, data } = req.fetch;
+  let url, options;
+  switch (action) {
+    case 'getAvailabilityForDocket':
+      url = searchParamsURL({
+        base: courtListenerURL('dockets'),
+        params: {
+          ...data,
+          source__in: '1,3,5,7,9,11,13,15',
```

Comment:
I think we need to update the sources here according to the new values added in:
https://github.com/freelawproject/courtlistener/blob/main/cl/search/docket_sources.py

so the docket lookup can be successful in all cases?